agent/engine/docker_task_engine.go

```diff
@@ -771,6 +771,11 @@ func (engine *DockerTaskEngine) pullAndUpdateContainerReference(task *api.Task,
 		defer container.SetRegistryAuthCredentials(credentials.IAMRoleCredentials{})
 	}
 
+	//if using asm auth, tag the asm authData with taskID
+	if container.IsASMImagePullAuth() {
```

Comment:
It seems like `Task.PostUnmarsalTask()` is the method where this should be happening. Any specific reason for doing this here?

Reply:
hm i think you're right since it's just adding metadata to the ASMAuthData struct. i put it here cause we branch the logic for setting pull auth credentials if using ecr and execution role. 

but yea, post unmarshall definitely makes more sense in this case.

Reply:
let's move it there then. you might not have to deal with https://github.com/adnxn/amazon-ecs-agent/pull/3/files/86a4b557405523ed9dafa8d45b20ecfc1b8f6666#diff-f7f8fe813f3277a629bc1d76f2919366R315 at all if you do that i think